ui/admin/app/routes/onboarding.js

```diff
@@ -1,36 +1,122 @@
+  async beforeModel() {
+    const orgs = await this.store.query('scope', {
+      scope_id: 'global',
+    });
+    if (!this.session.isAuthenticated || orgs.length)
+      this.router.transitionTo('index');
   }
 
   /**
    * List all scopes (orgs) under global.
+   * Creare org, project, target and roles
    */
-  model() {
-    return this.store.query('scope', {
-      scope_id: 'global',
-    });
+  async model() {
+    return {
+      org: this.store.createRecord('scope', {
+        type: 'org',
+        scopeID: 'global',
+        name: 'SecOps',
+        description: 'Sample org created by quick setup',
+      }),
+      project: this.store.createRecord('scope', {
+        type: 'project',
+        name: 'AWS',
+        description: 'Sample project created by quick setup',
+      }),
+      target: this.store.createRecord('target', {
+        type: TYPE_TARGET_TCP,
+        name: 'Test target',
+        description: 'Sample target created by quick setup',
+      }),
+      role: this.store.createRecord('role', {
+        name: 'test_target_role',
+      }),
+    };
+  }
+
+  // =actions
+  @action
+  @loading
+  @notifyError(function () {
+    return this.intl.t('errors.onboarding-failed.description');
+  })
+  async createTarget(targetAddress, targetPort) {
+    // Create org and Project
+    try {
+      await this.createOrgAndProject();
+    } catch (error) {
+      throw new Error();
+    }
+
+    // Create target
+    try {
+      await this.createSampleTarget(targetAddress, targetPort);
+      // Redirect to success
+      this.router.transitionTo('onboarding.success');
+    } catch (error) {
+      // Redirect to orgs
+      this.router.replaceWith('scopes.scope', 'global');
+      throw new Error();
+    }
   }
 
   /**
-   * If any orgs already exist, redirect to index to skip onboarding.
+   * Persist org and project previosly created in model()
    */
-  redirect(model) {
-    // If there are orgs already created, redirect to index.
-    if (model.length) this.router.transitionTo('index');
+  async createOrgAndProject() {
+    const { org, project } = this.currentModel;
+
+    await org.save();
+    project.scopeID = org.id;
+    await project.save();
+  }
+
+  /**
+   * Creates a Target with provided address and port
+   * @param {string} targetAddress
+   * @param {string} targetPort
+   */
+  async createSampleTarget(targetAddress, targetPort) {
+    const { target, role, project, org } = this.currentModel;
+
+    // Format target and persist it
+    target.scopeID = project.id;
+    target.address = targetAddress;
+    target.default_port = targetPort;
+    await target.save();
+
+    // Format role and role grants and persist them
+    role.scopeID = org.id;
+    role.grant_scope_id = project.id;
+    await role.save();
+    await role.saveGrantStrings([
+      `type=target;actions=list`,
+      `id=${target.id};actions=authorize-session`,
+    ]);
+  }
+
+  @action
+  @loading
```

Comment:
Is this `loading` decorator needed for a transition only?

Reply:
You are 100% right, that is not need at all here, deleted.